### PR TITLE
Add boundary condition support to Brick domain creator

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Add boundary condition support to Brick domain creator

Note: I didn't scope all the checks in the test because I can't spare the time and it's not really a very important (or at all important) thing to do.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Domain/Creators/Test_Brick.cpp [hunk 1]
```diff
@@ -213,6 +253,23 @@ void test_brick() {
            {Direction<3>::upper_zeta(), {0, aligned_orientation}}}},
       std::vector<std::unordered_set<Direction<3>>>{{}});
 
+  const creators::Brick periodic_brick_boundary_coitions{
```
Comment (kidder) [inline tests/Unit/Domain/Creators/Test_Brick.cpp]: coitions -> conditions

**Review by kidder — CHANGES_REQUESTED** [review]

**Review by nilsvu — DISMISSED** [review]

**Comment by nilsvu** [comment]
no need to get my re-approval when squashing in the minor change above

File: src/Domain/Creators/Brick.hpp [hunk 2]
```diff
@@ -79,17 +81,46 @@ class Brick : public DomainCreator<3> {
...
   };
 
-  using options =
-      tmpl::list<LowerBound, UpperBound, IsPeriodicIn, InitialRefinement,
-                 InitialGridPoints, TimeDependence>;
+  template <typename BoundaryConditionsBase>
+  struct BoundaryCondition {
+    static std::string name() noexcept { return "BoundaryCondition"; }
+    static constexpr Options::String help =
+        "The boundary condition to impose on all sides.";
```
Comment (nilsvu) [inline src/Domain/Creators/Brick.hpp:88]: Since it's fairly straight-forward to add support for different boundary conditions on either side, how about we make these a bunch of `good first issue`s (also for the other domain creators that have "not yet implemented" asserts)? :)

Comment (nilsdeppe) [reply src/Domain/Creators/Brick.hpp:88]: Added #2871

**Comment by nilsdeppe** [comment]
Okay, squashed in the rename and added the issue. Thanks for the reviews!

**Review by fmahebert — APPROVED** [review]

**Review by kidder — APPROVED** [review]
